### ISSUE TITLE
(Bug) (Windows) `wally install` fails with `error loading known-hosts: ; class-Ssh (23)`

### ISSUE THREAD

**Issue by zachshattuck** [UpliftGames/wally#190]
Overview
---
Hey all, I'm fairly new to Roblox development and am setting up a basic Aftman/Rojo/Wally project.
When trying to run `wally install` after adding a package to the .toml file, I am given the following error message:
```
> wally install
Error cloning Git repository https://github.com/UpliftGames/wally-index

Caused by:
    error loading known_hosts: ; class=Ssh (23)
```

My `known_hosts` file exists in the standard location (`~/.ssh/known_hosts`) and I am able to clone [UpliftGames/wally-index](https://github.com/UpliftGames/wally-index) via SSH normally.

After some digging, it seems like it's likely related to [this known issue in `git2-rs` versions 0.16 and below,](https://github.com/rust-lang/git2-rs/issues/937) - classic Windows pathname weirdness. I see that the current version of wally uses `git2@0.16.2`. It's possible that just updating to v0.17.0 will fix this.

Any help would be appreciated!

More information
---
aftman 0.3.0
Rojo 7.4.4
wally 0.3.2
Windows 11 Home v10.0.22631 Build 22631  

wally.toml:
```toml
[package]
name = "zdogs/roblox-game-1"
version = "0.1.0"
registry = "https://github.com/UpliftGames/wally-index"
realm = "shared"

[dependencies]
React = "jsdotlua/react@17.0.0"
```

**Comment by zachshattuck** [comment]
Just an update - as a workaround, removing a line in my git config that rewrites HTTPS to SSH does fix this issue and everything installs correctly.

I did a custom local build of wally that uses `git2@0.19.0` and it fixes the original error, but presents a new one:
```
> ..\wally\target\release\wally.exe install
[INFO ] Updating package index git@github.com:UpliftGames/wally-index...
could not update package index

Caused by:
    0: could not fetch Git repository
    1: no authentication available
```

**Comment by Dekkonot** [comment]
SSH support isn't really present right now. There's a PR to add it that's unmerged though.

If you build off of https://github.com/UpliftGames/wally/pull/177 does it resolve your problem?

**Comment by zachshattuck** [comment]
Oh I see, thanks for the info!
That branch didn't work, but it did seem to get farther along, and I got a *different* error message.

```
..\wally\target\release\wally.exe install
[INFO ] Updating package index git@github.com:UpliftGames/wally-index...
could not update package index

Caused by:
    0: could not fetch Git repository
    1: remote rejected authentication: Failed getting response; class=Ssh (23); code=Auth (-16)
```

I went exploring again for more information and it appears that this is a another [known issue](https://github.com/libgit2/libgit2/issues/4958#issuecomment-457440579) with [`libssh2`](https://github.com/libgit2/libgit2/issues/4958#issuecomment-457440579), which [`libgit2`](https://github.dev/libgit2/libgit2) (and therefore [`git2-rs`](https://github.com/rust-lang/git2-rs)) uses for SSH. `libssh2` is hardcoded to use Pageant for Windows.

I actually had the same issue using SSH git cloning when running `cargo build`, and I had to set a global cargo config option for it to work (see [`net.get_fetch_with_cli`](https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli) and [SSH Authentication](https://doc.rust-lang.org/cargo/appendix/git-authentication.html#ssh-authentication) of the Cargo Book). It runs [`fetch_with_cli`](https://github.com/rust-lang/cargo/blob/f95eeb66200adbbd6d9e84e551c5a06f3340b316/src/cargo/sources/git/utils.rs#L1185), which spawns `git` and fetches that way.


TLDR; this whole thing might just be because I'm on Windows trying to use SSH authentication 😂. Sorry for the link-heavy post.

**Comment by Dekkonot** [comment]
As someone who is _also_ on Windows and uses SSH authentication for git, I totally get it haha.

libssh2 being hardcoded to use a particular agent is frustrating. I'll take a look when I have some time to see if mimicking how Cargo does this is feasible.

**Comment by zachshattuck** [comment]
Awesome, thank you so much for your help! I have some free time at the moment, so I'm also poking currently poking around on a branch to see if I can get something like what Cargo does working. I'll keep you updated.
